decimal.go

```diff
@@ -1022,6 +1022,10 @@ func (d Decimal) MarshalJSON() ([]byte, error) {
 // UnmarshalBinary implements the encoding.BinaryUnmarshaler interface. As a string representation
 // is already used when encoding to text, this method stores that string as []byte
 func (d *Decimal) UnmarshalBinary(data []byte) error {
+	if len(data) < 4 {
+		return fmt.Errorf("error decoding binary %v: too short", data)
```

Comment:
It could be only me, but I think an error message should be more verbose. I think we should include a note that data should be at least 5 bytes long for proper unmarshaling. 
What do you think?